### PR TITLE
Migrate iscsi API group to library model

### LINKED ISSUE #238: Refactor the iSCSI API group to become a library

### PR DESCRIPTION
**Pull request by alexander-ding**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #238 

**Special notes for your reviewer**:
Removed the tests for the v1 API since it doesn't follow the versioned API like other API groups, which makes having two sets of tests a lot more work. We're removing v1 soon anyways so this is probably fine.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
Deferring release note to the version bump PR.

/cc @mauriciopoppe

### CODE REVIEW

File: pkg/disk/disk.go [hunk 1]
```diff
@@ -17,13 +17,29 @@ type Interface interface {
 	DiskStats(context.Context, *DiskStatsRequest) (*DiskStatsResponse, error)
 	GetAttachState(context.Context, *GetAttachStateRequest) (*GetAttachStateResponse, error)
 	GetDiskNumberByName(context.Context, *GetDiskNumberByNameRequest) (*GetDiskNumberByNameResponse, error)
+
+	// GetDiskState gets the offline/online state of a disk.
```
Comment (mauriciopoppe) [inline pkg/disk/disk.go:21]: I think this could be part of a different PR, I'd limit this change to iSCSI changes only

File: vendor/github.com/kubernetes-csi/csi-proxy/client/groups/iscsi/v1alpha2/client_generated.go [hunk 2]
```diff
@@ -1,98 +0,0 @@
-// Code generated by csi-proxy-api-gen. DO NOT EDIT.
```
Comment (mauriciopoppe) [inline vendor/github.com/kubernetes-csi/csi-proxy/client/groups/iscsi/v1alpha2/client_generated.go:1]: Files are getting removed here while in previous PRs it was only about adding changes, is this intended or should this happen later?

File: pkg/disk/disk.go (hunk 1 above)
Comment (alexander-ding) [reply pkg/disk/disk.go:21]: Oh oops I meant to leave these changes for later. Accidentally pushed them.

File: vendor/github.com/kubernetes-csi/csi-proxy/client/groups/iscsi/v1alpha2/client_generated.go (hunk 2 above)
Comment (alexander-ding) [reply vendor/github.com/kubernetes-csi/csi-proxy/client/groups/iscsi/v1alpha2/client_generated.go:1]: Yes that's intended. See the note
> Removed the tests for the v1 API since it doesn't follow the versioned API like other API groups, which makes having two sets of tests a lot more work. We're removing v1 soon anyways so this is probably fine.
This means that none of the pkg code uses the v1 client anymore, so it causes the vendor dependency to be deleted.

Comment (mauriciopoppe) [reply vendor/github.com/kubernetes-csi/csi-proxy/client/groups/iscsi/v1alpha2/client_generated.go:1]: ack

**Comment by mauriciopoppe** [comment]
I think you could try squashing your changes and then we can merge it

**Comment by alexander-ding** [comment]
> I think you could try squashing your changes and then we can merge it

Done! Thanks for the review :)

**Review by mauriciopoppe — COMMENTED** [review]
/lgtm
/approve
